mne/io/eeglab/eeglab.py

```diff
@@ -606,16 +610,16 @@ def _read_eeglab_events(eeg, event_id=None, event_id_func='strip_to_integer'):
         except (ValueError, TypeError):  # if event_id_func fails
             pass  # We're already raising warnings above, so we just drop
 
-    if len(events) < len(types):
-        warn("Some event codes could not be mapped to integers. Use the "
-             "`event_id` parameter to map such events to integers manually.")
-    if len(events) < 1 and len(types) > 0:
+    if len(events) < 1:
         # warn if events exist, but none found
         warn("No events could be converted, consider adding an `event_id`."
              " As is, the trigger channel will consist entirely of zeros.")
         return np.zeros((0, 3))
-    else:
-        return np.asarray(events)
+    elif len(events) < len(types):
+        warn("Some event codes could not be mapped to integers. Use the "
+             "`event_id` parameter to map such events to integers manually.")
```

Comment:
Really? I think that's a difference - one means you've managed to convert *something* and missed a bit, the other means all of your events are missed.